File: Java/Loop/src/Fila.java
import java.time.Duration;
import java.time.LocalDateTime;
import java.util.LinkedList;
import java.util.Stack;

public class Fila {
    public static void main(String[] args) {

        LinkedList<Long> fila = new LinkedList<>();
        LocalDateTime horaDeInicio;
        horaDeInicio = LocalDateTime.now();
        for (long i = 0; i < 1000000; i++) {
            fila.push(i);
        }
        System.out.println("Horario de finalizacaoo:" + Duration.between(horaDeInicio, LocalDateTime.now()).abs().toString().replace("S", "").replace("PT", ""));
        System.out.println("Uso de memoria: " + (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()) / (1024 * 1024) + " MB");

        fila.clear();
        System.gc();
        horaDeInicio = LocalDateTime.now();
        for (long i = 0; i < 50000000; i++) {
            fila.push(i);
        }
        System.out.println("Horario de finalizacaoo:" + Duration.between(horaDeInicio, LocalDateTime.now()).abs().toString().replace("S", "").replace("PT", ""));
        System.out.println("Uso de memoria: " + (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()) / (1024 * 1024) + " MB");

        fila.clear();
        System.gc();
        horaDeInicio = LocalDateTime.now();
        for (long i = 0; i < 100000000; i++) {
            fila.push(i);
        }
        System.out.println("Horario de finalizacaoo:" + Duration.between(horaDeInicio, LocalDateTime.now()).abs().toString().replace("S", "").replace("PT", ""));
        System.out.println("Uso de memoria: " + (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()) / (1024 * 1024) + " MB");

        fila.clear();
        System.gc();
        horaDeInicio = LocalDateTime.now();
        for (long i = 0; i < 150000000; i++) {
            fila.push(i);
        }
        System.out.println("Horario de finalizacaoo:" + Duration.between(horaDeInicio, LocalDateTime.now()).abs().toString().replace("S", "").replace("PT", ""));
        System.out.println("Uso de memoria: " + (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()) / (1024 * 1024) + " MB");

        fila.clear();
        System.gc();
        horaDeInicio = LocalDateTime.now();
        for (long i = 0; i < 200000000; i++) {
            fila.push(i);
        }
        System.out.println("Horario de finalizacaoo:" + Duration.between(horaDeInicio, LocalDateTime.now()).abs().toString().replace("S", "").replace("PT", ""));
        System.out.println("Uso de memoria: " + (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()) / (1024 * 1024) + " MB");
    }
}

File: Python/Fila.py
import datetime
import sys
from collections import deque

fila = deque()
horaDeInicio = datetime.datetime.now()
for i in range(1000000):
    fila.append(i)
print("Horario de finalizacao:", (datetime.datetime.now() - horaDeInicio).total_seconds())
print("Uso de memoria: {:.0f} MB".format((sys.getsizeof(fila)/1024)/1024))

fila.clear()
horaDeInicio = datetime.datetime.now()
for i in range(50000000):
    fila.append(i)
print("Horario de finalizacao:", (datetime.datetime.now() - horaDeInicio).total_seconds())
print("Uso de memoria: {:.0f} MB".format((sys.getsizeof(fila)/1024)/1024))

fila.clear()
horaDeInicio = datetime.datetime.now()
for i in range(100000000):
    fila.append(i)
print("Horario de finalizacao:", (datetime.datetime.now() - horaDeInicio).total_seconds())
print("Uso de memoria: {:.0f} MB".format((sys.getsizeof(fila)/1024)/1024))

fila.clear()
horaDeInicio = datetime.datetime.now()
for i in range(150000000):
    fila.append(i)
print("Horario de finalizacao:", (datetime.datetime.now() - horaDeInicio).total_seconds())
print("Uso de memoria: {:.0f} MB".format((sys.getsizeof(fila)/1024)/1024))

fila.clear()
horaDeInicio = datetime.datetime.now()
for i in range(200000000):
    fila.append(i)
print("Horario de finalizacao:", (datetime.datetime.now() - horaDeInicio).total_seconds())
print("Uso de memoria: {:.0f} MB".format((sys.getsizeof(fila)/1024)/1024))


